# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/test/chrome/test_bug682305.html
# Commit: 74f99033251c
# Full Hash: 74f99033251cd6c36eb6b0b5a4713eb6b5c793fa
# Author: Jonathan Kingston <jkt@mozilla.com>
# Date: 2019-02-13 09:48:26
# Regressor Bug: 1520868
# File Overlap Count: 1
# Description:
#   Bug 1520868 - Replacing AsyncOpen2 with AsyncOpen always r=valentin
#   
#   Replacing js and text occurences of asyncOpen2
#   Replacing open2 with open
#   
# ==============================================================================

diff -r 3378fd5f4f22 -r 74f99033251c dom/base/test/chrome/test_bug682305.html
--- a/dom/base/test/chrome/test_bug682305.html	Tue Feb 12 17:58:19 2019 +0100
+++ b/dom/base/test/chrome/test_bug682305.html	Tue Feb 12 16:08:25 2019 +0000
@@ -52,34 +52,30 @@
   loadGroup: null,
   name: null,
   status: Cr.NS_OK,
-  asyncOpen(listener, context) {
+  asyncOpen(listener) {
+    // throws an error if security checks fail
+    var outListener = contentSecManager.performSecurityCheck(this, listener);
     let stream = this.open();
+    let context = null;
     try {
-      listener.onStartRequest(this, context);
-    } catch (e) {}
-    try {
-      listener.onDataAvailable(this, context, stream, 0, stream.available());
+      outListener.onStartRequest(this, context);
     } catch (e) {}
     try {
-      listener.onStopRequest(this, context, Cr.NS_OK);
+      outListener.onDataAvailable(this, context, stream, 0, stream.available());
+    } catch (e) {}
+    try {
+      outListener.onStopRequest(this, context, Cr.NS_OK);
     } catch (e) {}
   },
-  asyncOpen2(listener) {
+  open() {
     // throws an error if security checks fail
-    var outListener = contentSecManager.performSecurityCheck(this, listener);
-    return this.asyncOpen(outListener, null);
-  },
-  open() {
+    contentSecManager.performSecurityCheck(this, null);
+
     let data = "bar";
     let stream = Cc["@mozilla.org/io/string-input-stream;1"].createInstance(Ci.nsIStringInputStream);
     stream.setData(data, data.length);
     return stream;
   },
-  open2() {
-    // throws an error if security checks fail
-    contentSecManager.performSecurityCheck(this, null);
-    return this.open();
-  },
   isPending() {
     return false;
   },